### PR TITLE
ROX-19617: Update policy info

### PR DESCRIPTION
**Pull request by kcarmichael08**
Version(s):

4.4+

Issues:

https://issues.redhat.com/browse/ROX-19617

Links to docs previews:

https://77200--ocpdocs-pr.netlify.app/openshift-acs/latest/operating/default-security-policies.html
https://77200--ocpdocs-pr.netlify.app/openshift-acs/latest/operating/manage-security-policies.html#policy-criteria_manage-security-policies

QE review:
- [ ] QE has approved this change.

Additional information:

Technical reviewers: Please see question in comments at the end of PR - thank you!

### CODE REVIEW

File: modules/policy-criteria.adoc [hunk 1]
```diff
@@ -312,7 +312,8 @@ Examples: +
 
 
 | Container Memory Request
-| Check for the amount of memory reserved for a given resource.
+| Number, including fraction, of MB requested.
+// Do we convert the K8s resource into MB (including convert MiB to MB) ? If so, the documentation should explain this conversion .
```
Comment (kcarmichael08) [inline modules/policy-criteria.adoc:317]: Question for tech reviewers here ^^^

File: modules/policy-criteria.adoc [hunk 2]
```diff
@@ -279,6 +279,7 @@ AND, OR
 | RAW=key=value to match an environment variable as directly specified in the deployment configuration with a specific key and value. `value` can be omitted to match on only the key. +
 
 If the environment variable is not directly defined in the configuration, then the format SOURCE=KEY can be used, where SOURCE is one of SECRET_KEY, CONFIG_MAP_KEY, FIELD or RESOURCE_FIELD. In this case, criteria can only match the key and not the value.
+//Does the drop-down in the GUI mean that you can choose to use the values that are contained in these fields? And if so, where are these fields located? In config files or somewhere else? Have been requested to add additional information & clarification to this criteria.
```
Comment (kcarmichael08) [inline modules/policy-criteria.adoc:282]: Question for tech reviewers here ^^^

File: modules/policy-criteria.adoc [hunk 3]
```diff
@@ -274,11 +274,19 @@ AND, OR
...
+| Check environment variables by name or value. When you create a policy that includes the environment variable attribute,
+you can choose which types of environment variables the policy should match. For example, you can specify raw values, which are provided directly in the deployment YAML, or you can specify references to values from ConfigMaps, Secrets, fields, or resource requests or limits.
+For any type other than a raw value specified directly in the deployment YAML, the corresponding `value` attribute of the policy rule is ignored. In this case, the policy match is evaluated on the existence of the specified environment variable type. Additionally, this criteria disallows the creation of policies with a non-empty `value` attribute for types other than raw values.
 | Environment Variable
-| RAW=key=value to match an environment variable as directly specified in the deployment configuration with a specific key and value. `value` can be omitted to match on only the key. +
+a| RAW=key=value to match an environment variable as directly specified in the deployment configuration with a specific key and value. `value` can be omitted to match on only the key. +
```
Comment (clickboo) [inline modules/policy-criteria.adoc:281]: I'd use deployment configuration or deployment YAML consistently. Pick one - I really can't say which one is better but if I were to guess - deployment configuration? (deployment YAML speaks to an engineer though :-))

File: modules/policy-criteria.adoc [hunk 4]
```diff
@@ -274,11 +274,19 @@ AND, OR
...
+
+If the environment variable is not directly defined in the configuration, then the format SOURCE=KEY can be used, where SOURCE is one of the following objects. In this list, the API object label is provided first; the label in the user interface is provided in parentheses.
```
Comment (clickboo) [inline modules/policy-criteria.adoc:283]: I'd add the list after: following objects.  And then continue "In the above list, the API object label ...." after the bulleted list.

File: modules/policy-criteria.adoc (hunk 1 above)
Comment (clickboo) [reply modules/policy-criteria.adoc:324]: I will have to check.

**Review by clickboo — COMMENTED** [review]
Few nits.

File: modules/policy-criteria.adoc [hunk 5]
```diff
@@ -330,25 +339,25 @@ Examples: +
...
 | *Deploy*, +
 *Runtime* (when used with a Runtime criterion)
 
 
 | Root filesystem writeability
-| Containers running with the root file system configured as read only.
+| Check if a deployment is configured in `readOnlyFilesystem` mode.
 | Read-Only Root Filesystem
-| Boolean
+| Boolean: `true` when `readOnlyRootFilesystem` in the respective `PodSecurityContext` is set to `true`
```
Comment (shipsing) [inline modules/policy-criteria.adoc:353]:
```suggestion
| Boolean: `true` when the value of the `readOnlyRootFilesystem` field in the respective `PodSecurityContext` is set to `true`
```

File: modules/policy-criteria.adoc [hunk 6]
```diff
@@ -330,25 +339,25 @@ Examples: +
...
-| Boolean
+| Boolean: `true` when the `privileged` field in the respective `PodSecurityContext` is set to `true`
 | ✕
 | *Deploy*, +
 *Runtime* (when used with a Runtime criterion)
 
 
 | Root filesystem writeability
-| Containers running with the root file system configured as read only.
+| Check if a deployment is configured in `readOnlyFilesystem` mode.
```
Comment (shipsing) [inline modules/policy-criteria.adoc:351]:
```suggestion
| Check if a deployment is configured in the `readOnlyFilesystem` mode.
```

File: modules/policy-criteria.adoc [hunk 7]
```diff
@@ -330,25 +339,25 @@ Examples: +
 
 
 | Privileged container
-| Privileged running deployments.
+| Check if a deployment is configured in privileged mode. This criterion only checks against the value of the `privileged` field in the respective link:https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#securitycontext-v1-core[Pod Security Context].
 | Privileged Container
-| Boolean
+| Boolean: `true` when the `privileged` field in the respective `PodSecurityContext` is set to `true`
```
Comment (shipsing) [inline modules/policy-criteria.adoc:344]:
```suggestion
| Boolean: `true` when the value of the `privileged` field in the respective `PodSecurityContext` is set to `true`
```

File: modules/policy-criteria.adoc [hunk 8]
```diff
@@ -330,25 +339,25 @@ Examples: +
 
 
 | Privileged container
-| Privileged running deployments.
+| Check if a deployment is configured in privileged mode. This criterion only checks against the value of the `privileged` field in the respective link:https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#securitycontext-v1-core[Pod Security Context].
```
Comment (shipsing) [inline modules/policy-criteria.adoc:342]:
```suggestion
| Check if a deployment is configured in `privileged` mode. This criterion only checks the value of the `privileged` field in the respective link:https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#securitycontext-v1-core[Pod Security Context].
```
Can we link to the upstream Kubernetes doc??

Comment (shipsing) [reply modules/policy-criteria.adoc:342]: is `privileged` a mode or just a field?? need to maintain consistency for clarity.

File: modules/policy-criteria.adoc [hunk 9]
```diff
@@ -274,11 +274,19 @@ AND, OR
...
 | Environment Variable
-| RAW=key=value to match an environment variable as directly specified in the deployment configuration with a specific key and value. `value` can be omitted to match on only the key. +
+a| RAW=key=value to match an environment variable as directly specified in the deployment configuration with a specific key and value. `value` can be omitted to match on only the key. +
+
+If the environment variable is not directly defined in the configuration, then the format SOURCE=KEY can be used, where SOURCE is one of the following objects. In this list, the API object label is provided first; the label in the user interface is provided in parentheses.
+
+* SECRET_KEY (SecretKeyRef)
+* CONFIG_MAP_KEY (ConfigMapRef)
+* FIELD (FieldRef)
+* RESOURCE_FIELD (ResourceFieldRef)
```
Comment (shipsing) [inline modules/policy-criteria.adoc:288]: Suggestion:
In the preceding list, you can view the `SOURCE` format first includes the API object label and then the user interface label in parentheses.

**Comment by shipsing** [comment]
@kcarmichael08, Nice work! provided some suggestions for your consideration. Thanks

File: modules/policy-criteria.adoc (hunk 3 above)
Comment (kcarmichael08) [reply modules/policy-criteria.adoc:281]: Our style guide is against "deployment configuration" so I think YAML is better.

File: modules/policy-criteria.adoc [hunk 10]
```diff
@@ -330,25 +339,25 @@ Examples: +
...
-| Boolean
+| Boolean: `true` when `readOnlyRootFilesystem` in the respective `PodSecurityContext` is set to `true`
 | ✕
 | *Deploy*, +
 *Runtime* (when used with a Runtime criterion)
 
 
 | Seccomp Profile Type
-| The type of seccomp profile allowed for the container.
+| The type of `seccomp` profile defined for the deployment. If `seccomp` options are provided at both the pod & container level, the container options override the pod options. See link:https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context-1[Security Context].
```
[the remark this answers is not in the document]
Comment (kcarmichael08) [reply modules/policy-criteria.adoc:360]: As far as I know, it's OK in the ACS branch.

File: modules/policy-criteria.adoc (hunk 8 above)
Comment (kcarmichael08) [reply modules/policy-criteria.adoc:342]: Privileged is a mode, which is configured by having a value of "true" in the "privileged" field, so I think this is OK?

**Pull request merged**